suites/gen3client/gen3clientTest.js

```diff
@@ -0,0 +1,174 @@
+// 3. using the creds.json, configure the profile
+// 4. upload a file
+// 5. check the indexd record
+// 6. download the file
+
+// this test will run in cdis-data-client and gen3-qa PRs
+
+// to run this locally please ensure golang is installed in the machine (runs on linux and mac)
+
+Feature('Gen3-client @requires-fence @requires-indexd @gen3-client');
+
+const { expect } = require('chai');
+const { output } = require('codeceptjs');
+const fs = require('fs');
+const { execSync } = require('child_process');
+const users = require('../../utils/user');
+const { sleepMS } = require('../../utils/apiUtil');
+
+const I = actor();
+I.cache = {};
+
+BeforeSuite(async ({ I, files}) => {
+  I.cache.fileData = 'This is a test file uploaded via gen3-client test';
+  const userAccessToken = users.indexingAcct.accessToken;
+  I.cache.accessToken = userAccessToken;
+  output.debug(`Access token: ${I.cache.accessToken}`);
+
+  // create a file that can be uploaded
+  const fileToBeUploaded = `file_${Date.now()}.txt`;
+  const filePath = `./${fileToBeUploaded}`;
+  console.log(`File to be uploaded : ${fileToBeUploaded}`);
+  await files.createTmpFile(filePath, I.cache.fileData);
+  I.cache.fileToBeUploaded = fileToBeUploaded;
+
+  // Install gen3-client
+  I.cache.goPath = execSync('go env GOPATH', { shell: 'bash' }).toString().trim();
+  output.debug(`#### goPath: ${I.cache.goPath}`);
+  output.debug(`#### REPO: ${process.env.REPO}`);
+  let clientInstallCommands = `mkdir -p ${I.cache.goPath}/src/github.com/uc-cdis` +
+      ` && cd ${I.cache.goPath}/src/github.com/uc-cdis` +
+      ' && git clone https://github.com/uc-cdis/cdis-data-client.git' +
+      ' && mv cdis-data-client gen3-client' +
+      ' && cd gen3-client' + 
+      ' && go get -d ./...' +
+      ' && go install .'
+  if(process.env.REPO === 'cdis-data-client' || process.env.REPO === 'gen3-client') {
+    output.debug(`#### change branch - ${process.env.CHANGE_BRANCH}`);
+    clientInstallCommands = `mkdir -p ${I.cache.goPath}/src/github.com/uc-cdis` +
+    ` && cd ${I.cache.goPath}/src/github.com/uc-cdis` +
+    ' && git clone https://github.com/uc-cdis/cdis-data-client.git' +
+    ' && mv cdis-data-client gen3-client' +
+    ' && cd gen3-client' +
+    ` && git checkout ${process.env.CHANGE_BRANCH}` +
+    ' && go get -d ./...' +
+    ' && go install .'
+  }
+
+  output.debug(`## clientInstallCommands - ${clientInstallCommands}`)
+  execSync(clientInstallCommands, { shell: 'bash' });
+
+  const version = execSync(`${I.cache.goPath}/bin/gen3-client --version`, { shell: 'bash' });
+  output.debug(`### gen3-client version installed is ${version}`);
+});
+
+AfterSuite(async ({ I, files, indexd }) => {
+  // delete the fileUploaded and creds.file
+  output.log('Deleting test upload files ...');
+  files.deleteFile(I.cache.fileToBeUploaded);
+  output.log('Deleting creds.file ...');
+  files.deleteFile(I.cache.credsFile);
```

Comment:
log/profile/stats files for gen3-client exists in `~/.gen3`, you may want to clean it up as well, but also ok to leave as-is I think

Reply:
everything runs in the PR pod (not in the adminvm) which gets removed after the run, so cleanup is only needed if other tests get affected which isn't the case here I think.